client/components/empty-content/empty-content.jsx

```diff
@@ -32,12 +39,19 @@ module.exports = React.createClass( {
 		] ),
 		secondaryActionURL: React.PropTypes.string,
 		secondaryActionCallback: React.PropTypes.func,
+		mountEvent: React.PropTypes.string,
 		className: React.PropTypes.string,
 		isCompact: React.PropTypes.bool
 	},
 
 	componentDidMount: function() {
 		debug( 'Empty Content React component mounted.' );
+		if ( ! this.props.mountEvent ) {
+			return;
+		}
+		analytics.tracks.recordEvent( 'calypso_empty_content_component_impression', {
```

Comment:
I believe it's better to expose `onMount` than to make a UI component reliant on the analytics module.